src/App.tsx

```diff
@@ -1,14 +1,41 @@
-/* eslint-disable max-len */
-import React from 'react';
-import 'bulma/css/bulma.css';
-import '@fortawesome/fontawesome-free/css/all.css';
+import React, { useEffect, useState } from 'react';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { filterTodosByRule } from './services/service';
+
+import { Todo } from './types/Todo';
+
+import 'bulma/css/bulma.css';
+import '@fortawesome/fontawesome-free/css/all.css';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState<string>('');
+  const [rule, setRule] = useState('all');
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [isModalLoading, setIsModalLoading] = useState(false);
+
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(res =>
+        res.filter(todo =>
+          todo.title.toLowerCase().includes(query?.toLowerCase()),
+        ),
+      )
+      .then(res => filterTodosByRule(res, rule))
+      .then(setTodos)
+      .finally(() => setIsLoading(false));
+  }, [query, rule]);
```

Comment:
No need to fetch todos every time. Just do it at once and simply keep filtered todos in a variable
```typescript
const filteredTodos = filterTodos(todos, rule);
```